src/ocean/task/Scheduler.d

```diff
@@ -615,6 +615,8 @@ final class Scheduler : IScheduler
     public void processEvents ( )
     {
         auto task = Task.getThis();
+        assert(task !is null);
```

Comment:
Why not `verify`?

Reply:
It will crash anyway even if check is missing

Reply:
Mm I don't get what you mean. Aren't these the possible cases in D2 builds?
* `assert`, `task is null`: crash.
* `assert`, `task !is null`: keeps running.
* `verify`, `task is null`: throws.
* `verify`, `task !is null`: keeps running.

Reply:
Few more options depending on `-release` but essentially yes. But don't forget that it crashes _right now_ if condition is false because there is no check at all. So this exactly matches assert purpose - turn crash / corruption into an easier debuggable crash.

Reply:
But didn't we go through and convert all such cases into `verify`s?

Reply:
We converted such _existing_ cases for _backwards compatibility_. That doesn't mean all them should really remain as such and definitely doesn't apply for new cases. See also https://github.com/sociomantic-tsunami/ocean/issues/289

Reply:
Yes, "crash with segfault" is that extra check in this case.

Reply:
It sounds like you're saying our guidelines are incorrect.

>Checks for programmer error / basic sanity or safety (e.g. pointer checks, integer expressions) should be performed using ocean's verify

Reply:
Ugh, yes. This statement is definitely incorrect. Checks for programmer error should always be done with `assert` unless you can't afford it to be removed. So, for example, protecting cast + raw memory access has to be done with `verify` because if you miss it, app will end up with memory corruption and that is very bad. But protecting null pointers with `verify` is almost always a waste, because it will crash the app immediately anyway.

Reply:
Well... the whole point of inventing `verify` was to avoid apps crashing due to an `assert` firing in D2.

Reply:
If that was the case, we have wasted a lot of effort on something fundamentally flawed. You can't avoid crashes no matter how hard you try and must embrace them :) From my PoV goal of `verify` is to protect against memory corruptions and similar damage.

Reply:
To word it differently - the fact that right now dhtnode can hit an `assert` (now `verify`) condition and survive after is a bug in dhtnode that I totally expect to be fixed some day, not a feature to support.

Reply:
>the fact that right now dhtnode can hit an assert (now verify) condition and survive after is a bug in dhtnode that I totally expect to be fixed some day

This is the key. Using `assert`s to notify us of such bugs causes crashes. Using `verify`s to notify us of such bugs (potentially) avoids crashes and allows us to log the bug. From the POV of a dhtnode, the latter is _way_ more desirable than the former.

Reply:
I disagree. Consider this PR - before my change it would always just segfault on wrong usage, in both D1 and D2. There are no asserts, just plain segfault. The very fact that it was working like that in deployed live apps for a very long time is a direct proof that this is perfectly fine.

This is reality that can't be worked around - no matter what policy you take on `assert`, your app will crash eventually and bugs will have to be fixed. Effort to make it as harmless as possible is meaningful, effort to prevent it altogether is not.

Reply:
>The very fact that it was working like that in deployed live apps for a very long time is a direct proof that this is perfectly fine.

It's only fine because it never came up. If it had have come up, an `assert` here would be as bad as a plain segfault, in terms of direct impact to the application.

>Effort to make it as harmless as possible is meaningful, effort to prevent it altogether is not.

This is exactly what I'm suggesting. Throwing is generally less harmful than crashing. Both have equal value in terms of notifying maintainers of bugs.